### DISCUSSION TITLE
Prioritise routes based on corners

### DISCUSSION THREAD

**Discussion by zeddash** [voxeltycoon/issues#511] [General]
As a precursor to priority roads/priority avoid roads, routes that are the same distance should be preferred with the least amount of turns. I think the current system seems to either prefer EW over NS or it tries to get closer to the destination as early as possible. I would rank routes based on turns then distance.

Red lines show the current route taken by iron ore trucks from bottom right (I changed the junction near the centre to force trucks up and not go through terminals), green is what I would expect trucks to take
![vt1](https://user-images.githubusercontent.com/46975001/71783619-387bb580-2fe1-11ea-8914-824d08ac8967.png)

**Comment by victorleblais** [comment]
I'm marking this as a bug even though the current route is normal according to the current implementation of the pathfinder.

In my opinion, it should indeed do what you say it should do.

There's something called the "Manhattan distance" which describes all the paths that are the same length on a grid, where you obviously have some with lots of turns and some with 1 or 2 turns.

It seems the games wants to take a route that is the closest to a straight line (ignoring roads) which makes it do more turns than what it should.

**Comment by hairintd** [comment]
it would be great if game will count other stations on the way when searching for path. The less stations - the better path. Actually, this option must have weight against length and turns count. 
Or implement overtake system.
